### PR TITLE
Feat(eos_designs,eos_cli_config_gen): Add metadata.platform to structured configuration

### PR DESCRIPTION
**Pull request by dlobato**
## Change Summary

Add metadata.platform to structured configuration

## Component(s) name

arista.avd.eos_designs
arista.avd.eos_cli_config_gen

## Proposed changes

- Add metadata.platform to schema for eos_cli_config_gen
- Output metadata.platform from eos_designs

## How to test

- All eos_designs molecule scenarios reran. No config changes.

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/input-variables.md [hunk 1]
```diff
@@ -579,6 +579,12 @@ roles/eos_cli_config_gen/docs/tables/management-cvx.md
 roles/eos_cli_config_gen/docs/tables/mcs-client.md
 --8<--
 
+### Metadata
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/input-variables.md]: I think we should move this section to the end of the doc and maybe add a line above the table snip explaining that this data is not generating any config etc.

**Review by ClausHolbechArista — APPROVED** [review]
A small comment on docs, but approving.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/input-variables.md (hunk 1 above)
Comment (dlobato) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/input-variables.md]: Moved to the bottom with a small paragraph describing the purpose of this variables

**Review by carlbuchmann — APPROVED** [review]
LGTM

**Pull request merged**